server/pulp/server/webservices/views/repositories.py

```diff
@@ -121,6 +121,35 @@ def _process_repos(repos, details, importers, distributors):
     return repos
 
 
+def _validate_repo_importer_existance(repo_id, importer_id):
+    """
+    Validates if the specified repo_id and importer_id are valid.
+    If not MissingResource exception is raised.
+
+    :param repo_id: id of the repo
+    :type repo_id: str
+    :param importer_id: id of the importer
+    :type importer_id: str
+
+    :return: key-value pairs describing the importer in use
+    :rtype:  dict
+    :raises pulp_exceptions.MissingResource: if repo or importer cannot be found
+    """
+
+    repo_query_manager = manager_factory.repo_query_manager()
+    repo = repo_query_manager.find_by_id(repo_id)
+    if repo is None:
+        raise pulp_exceptions.MissingResource(repo_id=repo_id)
```

Comment:
This works here but will fail when merging to master because of my recent mongoengine change. Here is what you will need to do in master (replacing the previous 4 lines):

``` python
from pulp.server.db import model

model.Repository.objects.get_repo_or_missing_resource(repo_id)
```